qiskit/transpiler/target.py

```diff
@@ -1043,24 +1043,25 @@ def target_to_backend_properties(target: Target):
                     continue
                 qubit = qargs[0]
                 props_list = []
-                if props.error is not None:
-                    props_list.append(
-                        {
-                            "date": datetime.datetime.utcnow(),
-                            "name": "readout_error",
-                            "unit": "",
-                            "value": props.error,
-                        }
-                    )
-                if props.duration is not None:
-                    props_list.append(
-                        {
-                            "date": datetime.datetime.utcnow(),
-                            "name": "readout_length",
-                            "unit": "s",
-                            "value": props.duration,
-                        }
-                    )
+                if props is not None:
+                    if props.error is not None:
+                        props_list.append(
+                            {
+                                "date": datetime.datetime.utcnow(),
+                                "name": "readout_error",
+                                "unit": "",
+                                "value": props.error,
+                            }
+                        )
+                    if props.duration is not None:
+                        props_list.append(
+                            {
+                                "date": datetime.datetime.utcnow(),
+                                "name": "readout_length",
+                                "unit": "s",
+                                "value": props.duration,
+                            }
+                        )
                 if not props_list:
                     qubit_props = {}
                     break
```

Comment:
This may be a too small nitpick in the original code, but these lines suggest we cannot mix ideal Measures and real Measures (if do that, real Measures will be silently converted to ideal Measures here, right?). Warning here (or allowing the mixture) might be nice. That said, I don't think of any practical use case of such a mixture, so I'm fine with keeping the code as is until we encounter any concrete issue.

Reply:
That's a good point, I do think it's unlikely in practice for there to be mixed ideal and real measurements but we should try to support that especially since it's possible combination and allowed in the target. The only reason we have to special case measurement here is because it's special in `BackendProperties`.